### PR TITLE
RDoc-1308 Document SQL Migration

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

**Comment by ml054** [comment]
TODO: update TOC once review is done.

File: Documentation/4.1/Raven.Documentation.Pages/studio/.docs.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[
+    {
+        "Path": "overview.markdown",
+        "Name": "Overview",
+        "Mappings": [],
+        "IsPlaceholder": "true"
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/.docs.json]: There is no longer need to use `IsPlaceholder`

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+4. **Export configuration** - Exports SQL configuration to file, so it can be finished later, by using *Continue existing migration option*. 
+
+5. **Fullscreen mode** - Toggles fullscreen mode. 
+
+6. **Settings menu** - Contains few settings options. 
+
+7. **Batch size** - Allows to change batch size of document insertion
+
+8. **Convert binary columns to attachments** - If this option is check, then binary columns are added to document as attachment, in other case binary content is inserted as string to target document.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: WRT as string - do we use base64 then?

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+5. **Fullscreen mode** - Toggles fullscreen mode. 
+
+6. **Settings menu** - Contains few settings options. 
+
+7. **Batch size** - Allows to change batch size of document insertion
+
+8. **Convert binary columns to attachments** - If this option is check, then binary columns are added to document as attachment, in other case binary content is inserted as string to target document. 
+
+9. **Partial migration** - Allows to limit amount of items being imported per each table. Please notice this option can introduce broken links, between documents.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: amount -> number

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  
+* **Many-to-one** relationship: Source table contains *foreign key*, target table contains *primary key*.
+
+  * Mode set to **skip** - Property is skipped
+
+  * Mode set to **link** - Property value contains single document identifier or null
+  
+  * Mode set to **embed** - Property value contains single nested document 
+
+{NOTE Embedded documents don't contain target table identifier as it is redundant. /}
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: target table identifier -> identifier ? I found "target table" confusing here

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+{PANEL/}
+
+{PANEL: Filtering / Transforming documents}
+
+Sql Migration allows to filter and/or transform rows being imported. Click on 'Filter/Transform' button to access those settings. 
+
+Flow goes like this:
+
+1. Rows are read from source SQL database by executing query defined in filter section.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: Important to underline that you can use `where` here (as it's show on the screenshot) which gives the ability to filter out as SQL statement level

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+Let's say we want to compute and store amount of ordered products per order. 
+
+![Figure 9.](images/sql-migration-transform.png "Computing 'ProductsCount' per order")
+
+We have 2 test modes available:
+
+- **Use first record matching query** - it runs the query defined in filter section and gets first result. 
+- **Specify primary key values to use** - user is presented with extra fields to provide value for each primary key column. 
+
+![Figure 10.](images/sql-migration-test-results.png "Test script results")
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: This section lacks the description of `throw 'skip'` to filter out at transformation script level

File: Documentation/4.1/Raven.Documentation.Pages/studio/.docs.json (hunk 1 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/.docs.json]: done

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown (hunk 2 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: done

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown (hunk 3 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: done

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown (hunk 4 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: done

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown (hunk 5 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: done

File: Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown (hunk 6 above)
Comment (ml054) [reply Documentation/4.1/Raven.Documentation.Pages/studio/database/settings/import-from-sql.markdown]: I will add this info to previous section, where I explain flow.

**Pull request merged**